### DISCUSSION TITLE
How to get a list of all channels ?

### DISCUSSION THREAD

**Discussion by hrutik7** [discordjs/discord.js#9890] [Q&A] [answered]
Hey I am new to discord.js will anyone please let me know How I will get the list of all channels?

**Comment by almostSouji** [comment, marked as answer]
The bot client as a [`.channels` manager](https://old.discordjs.dev/#/docs/discord.js/14.13.0/class/Client?scrollTo=channels) which exposes a [`.cache`](https://old.discordjs.dev/#/docs/discord.js/14.13.0/class/ChannelManager?scrollTo=cache) of all the channels in all the servers your bot is in. If you just want the channels of a specific server (guild), the [`Guild`](https://old.discordjs.dev/#/docs/discord.js/14.13.0/class/Guild?scrollTo=channels) structure behaves similarly.

**Comment by xero-lib** [comment]
@hrutik7 
```js
bot.channels.cache.forEach((channel) => console.log(channel.name));
```
will print every channel name currently in the bots cache.
